File: java/parser/src/main/java/gov/lbl/parser/parser/Base_prefixVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Base_prefix;

public class Base_prefixVisitor extends modelicaBaseVisitor<Base_prefix> {
    @Override
    public Base_prefix visitBase_prefix(modelicaParser.Base_prefixContext ctx) {
        Type_prefixVisitor type_prefixVisitor = new Type_prefixVisitor();
        String type_prefix = ctx.type_prefix().accept(type_prefixVisitor);
        return new Base_prefix(type_prefix);
    }
}

File: jsParser/parser/Base_prefixVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Base_prefix = require('../domain/Base_prefix');

const Type_prefixVisitor = require('./Type_prefixVisitor');

class Base_prefixVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitBase_prefix(ctx) {
        var type_prefix = "";

        if (ctx.type_prefix()) {
            var type_prefixVisitor = new Type_prefixVisitor.Type_prefixVisitor();
            type_prefix = type_prefixVisitor.visitType_prefix(ctx.type_prefix());
        }
                
        return new Base_prefix.Base_prefix(type_prefix);
    }
};

Base_prefixVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitBase_prefix = this.visitBase_prefix;
exports.Base_prefixVisitor = Base_prefixVisitor;

